packages/treasury/src/runLoC.js

```diff
@@ -0,0 +1,90 @@
+// @ts-check
+import { AmountMath } from '@agoric/ertp';
+import { handleParamGovernance, ParamType } from '@agoric/governance';
+import { Far } from '@agoric/marshal';
+import {
+  assertIsRatio,
+  assertProposalShape,
+  ceilMultiplyBy,
+  floorMultiplyBy,
+} from '@agoric/zoe/src/contractSupport/index.js';
+
+const { details: X, quote: q } = assert;
+
+export const CreditTerms = {
+  CollateralPrice: 'CollateralPrice',
+  CollateralizationRate: 'CollateralizationRate',
+};
+
+/**
+ * @param { ContractFacet } zcf
+ * @param {{ feeMintAccess: FeeMintAccess }} privateArgs
+ */
+const start = async (zcf, { feeMintAccess }) => {
+  const { main: initialValue } = zcf.getTerms();
+
+  const {
+    makePublicFacet,
+    makeCreatorFacet,
+    getParamValue,
+  } = handleParamGovernance(zcf, harden(initialValue));
```

Comment:
This is taking the creator's word for what the governed parameters should look like. 

The AMM declares its expectations in [params.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/vpool-xyk-amm/params.js), in a way that is used in the contract, and should have been used in the tests when creating  parameters for terms.

[stablecoinMachine](https://github.com/Agoric/agoric-sdk/blob/58c37780a506dfafc5356b3edcc8e13edfc537a6/packages/treasury/src/stablecoinMachine.js#L98) asserts that the governed params from terms match the local declaration to ensure that all and only the required params are present. If there are params that aren't used or verified at startup, this seems crucial to me.

Reply:
I don't understand how the `sameStructure` in stablecoinMachine works; it seems to be comparing the list `['ChargingPeriod', ...]` with something that looks like `[{name, type, value}, ...]`.

Reply:
They should both be `{ "vaultParams": ["ChargingPeriod", "RecordingPeriod", "InitialMargin", "LiquidationMargin", "InterestRate", "LoanFee"] }`, which comes directly from params.js.

When setting up the treasury in test-stablecoin, treasuryTerms contains `governedParams: governedParameterTerms`, where the latter is imported from params.